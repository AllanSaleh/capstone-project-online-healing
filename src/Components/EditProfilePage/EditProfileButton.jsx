import React from 'react';

const EditProfileButton = ({ text, bgColor, color, margin }) => (
  <button
    type="button"
    className={`${bgColor || 'bg-blue-dark'}
     text-xl text-black lg:${margin} my-2 rounded-lg uppercase px-12 py-3 lg:text-subtitle border transition duration-300 hover:bg-white hover:${`border-${color}`}
    hover:${`text-${color}`}`}
  >
    {text}
  </button>
);
export default EditProfileButton;
