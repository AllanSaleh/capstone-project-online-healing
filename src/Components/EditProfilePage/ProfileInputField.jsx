import React from 'react';

const ProfileInputField = ({ title, id, type, icon }) => (
  <div className="flex items-center my-6">
    <label htmlFor={id} className="text-subtitle capitalize inline-block w-1/3">
      {title}
    </label>
    <div className="w-2/3 inline-flex items-center border border-black border-opacity-20 rounded-lg pl-4">
      <input type={type} id={id} name={id} className="w-11/12 h-full outline-none py-3" />
      {icon && (
        <button type="button">
          <img src={icon} alt="profileIcon" />
        </button>
      )}
    </div>
  </div>
);

export default ProfileInputField;
