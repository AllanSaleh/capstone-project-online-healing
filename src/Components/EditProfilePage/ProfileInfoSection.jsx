import React from 'react';

import ProfileInputField from './ProfileInputField';
import BirthDateIcon from './Images/BirthDateIcon.svg';
import Plus from './Images/Plus.svg';

const ProfileInfoSection = ({ userProfile }) => (
  <form className="mb-8">
    <h1 className="text-3xl lg:text-title uppercase">Profile Info</h1>

    <ProfileInputField title="full name" type="text" id="full-name" value={userProfile?.fullname} />

    <div className="flex items-center my-6">
      <label
        htmlFor="education-level"
        className="text-sm md:text-lg lg:text-subtitle capitalize inline-block w-1/3"
      >
        education level
      </label>
      <div className="w-2/3 inline-flex items-center border border-black border-opacity-20 rounded-lg px-6 py-2">
        <select
          id="education-level"
          name="education-level"
          className="w-full h-full outline-none py-1"
        >
          <option value="1">bachelor</option>
          <option value="2">bachelor</option>
          <option value="3">bachelor</option>
          <option value="4">bachelor</option>
        </select>
      </div>
    </div>
    <ProfileInputField title="hobbies" type="text" id="hobbies" value={userProfile?.hobbies} />

    <div className="flex items-center my-6">
      <label
        htmlFor="family-members"
        className="text-sm md:text-lg lg:text-subtitle capitalize inline-block w-1/3"
      >
        family members
      </label>
      <div className="w-2/12 lg:w-1/12 inline-flex items-center border border-black border-opacity-20 rounded-lg px-4">
        <input
          type="text"
          id="family-members"
          name="family-members"
          className="w-full h-full outline-none py-3"
          value={userProfile?.family_size}
        />
      </div>
      <span className="text-sm md:text-lg lg:text-subtitle capitalize mx-4">member(s)</span>
    </div>

    <div className="flex items-center my-6">
      <label
        htmlFor="gender"
        className="sm:text-sm md:text-lg lg:text-subtitle capitalize inline-block w-1/3"
      >
        gender
      </label>
      <div className="w-2/3 inline-flex items-center border border-black border-opacity-20 rounded-lg px-6 py-2">
        <select id="gender" name="gender" className="w-full h-full outline-none py-1">
          <option value="1">Male</option>
          <option value="2">Female</option>
        </select>
      </div>
    </div>

    <ProfileInputField
      title="birth date"
      type="text"
      id="birth-date"
      icon={BirthDateIcon}
      value={userProfile?.birthdate}
    />
    <ProfileInputField title="email" type="email" id="email" value={userProfile?.email} />
    <ProfileInputField
      title="phone number"
      type="text"
      id="phone-number"
      value={userProfile?.phone_number}
    />
    <ProfileInputField title="upload ID" type="file" id="upload-id" icon={Plus} />
  </form>
);

export default ProfileInfoSection;
