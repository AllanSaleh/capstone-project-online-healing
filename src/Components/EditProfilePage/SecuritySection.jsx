import React from 'react';

import { useTranslation } from 'react-i18next';
import ProfileInputField from './ProfileInputField';
import LockIcon from './Images/LockIcon.svg';

export default function SecuritySection() {
  const { t } = useTranslation();
  return (
    <form className="my-16">
      <h1 className="text-3xl lg:text-title uppercase">{t('Edit.Security')}</h1>
      <ProfileInputField title={t('Edit.Pass')} type="password" id="password" icon={LockIcon} />
      <ProfileInputField
        title={t('Edit.Confirm')}
        type="password"
        id="confirm-password"
        icon={LockIcon}
      />
    </form>
  );
}
