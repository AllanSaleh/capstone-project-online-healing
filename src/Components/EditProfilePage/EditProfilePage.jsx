import React, { useState, useEffect } from 'react';
import { Link, useHistory } from 'react-router-dom';
import firebase from '../../firebase';

import ProfilePic from './Images/ProfilePic.svg';
import EditProfileButton from './EditProfileButton';
import ProfileInfoSection from './ProfileInfoSection';
import SecuritySection from './SecuritySection';

export default function EditProfilePage() {
  window.scrollTo(0, 0);
  const history = useHistory();
  const usersRef = firebase.firestore().collection('users');
  const [currentUser, setCurrentUser] = useState({});

  useEffect(() => {
    const loginStatus = JSON.parse(localStorage.getItem('loginStatus'));
    usersRef.get().then((querySnapshot) => {
      querySnapshot.forEach((doc) => {
        if (loginStatus === null) history.replace('/')
        else if (doc.data().user_id === loginStatus.user_id) setCurrentUser(doc.data());
      });
    });
  }, []);

  console.log(currentUser);
  return (
    <div className="px-sides pt-navbar">
      <h3 className="text-lg lg:text-subtitle text-red-500 w-full text-center">
        {!currentUser.complete &&
          'Please fill all the fields with correct and valid details to complete your profile.'}
      </h3>
      <div className="flex flex-col lg:flex-row">
        <div className="w-full p-8 lg:w-2/5 lg:p-4 lg:mx-2 text-center">
          <button type="button" className="hover:cursor-pointer">
            <img src={ProfilePic} alt="profile-pic" />
          </button>
        </div>
        <div className="w-full lg:w-3/5 p-4 mx-2">
          <ProfileInfoSection userProfile={currentUser} />
          <SecuritySection userProfile={currentUser} />
          <div className="flex flex-col lg:flex-row justify-between my-8">
            <EditProfileButton
              text="save changes"
              color="green-500"
              margin="mr-4"
              bgOpacity="bg-opacity-70"
            />
            <EditProfileButton
              text="delete account"
              color="red-500"
              margin="mx-4"
              bgOpacity="bg-opacity-70"
            />
            <EditProfileButton
              text="cancel"
              color="yellow-300"
              margin="ml-4"
              bgOpacity="bg-opacity-70"
            />
          </div>

          <div className="my-16">
            <h1 className="text-3xl lg:text-title capitalize mb-8 lg:mb-16">
              Payment Methods & Tickets
            </h1>
            <div className="text-center lg:text-left lg:inline-block mx-0 xl:mr-4">
              <h3 className="text-subtitle capitalize mb-2">
                {currentUser?.cards?.length} crads added
              </h3>
              <Link to="/SavedCards">
                <EditProfileButton text="show cards" color="blue-dark" />
              </Link>
            </div>
            <div className="text-center lg:text-left lg:inline-block mx-0 xl:ml-4">
              <h3 className="text-subtitle capitalize mb-2">
                {currentUser?.tickets} tickets remaining
              </h3>
              <Link to={{ pathname: '/', state: true }}>
                <EditProfileButton text="buy tickets" color="blue-dark" />
              </Link>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}
