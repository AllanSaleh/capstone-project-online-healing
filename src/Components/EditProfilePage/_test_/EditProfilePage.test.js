import * as React from 'react';
import * as renderer from 'react-test-renderer';
import EditProfilePage from '../EditProfilePage';

it('Edit Profile Page renders correctly', () => {
  const tree = renderer.create(<EditProfilePage />).toJSON();
  expect(tree).toMatchSnapshot();
});
