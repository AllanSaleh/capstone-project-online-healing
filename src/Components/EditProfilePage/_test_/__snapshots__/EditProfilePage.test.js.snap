// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit Profile Page renders correctly 1`] = `
<div
  className="px-sides pt-navbar"
>
  <h3
    className="text-subtitle text-red-500 w-full text-center"
  >
    Please fill all the fields with correct and valid details to complete your profile.
  </h3>
  <div
    className="flex"
  >
    <div
      className="w-2/5 p-4 mx-2 text-center"
    >
      <button
        className="hover:cursor-pointer"
        type="button"
      >
        <img
          alt="profile-pic"
          src="ProfilePic.svg"
        />
      </button>
    </div>
    <div
      className="w-3/5 p-4 mx-2"
    >
      <form
        className="mb-8"
      >
        <h1
          className="text-title uppercase"
        >
          Profile Info
        </h1>
        <div
          className="flex items-center my-6"
        >
          <label
            className="text-subtitle capitalize inline-block w-1/3"
            htmlFor="full-name"
          >
            full name
          </label>
          <div
            className="w-2/3 inline-flex items-center border border-black border-opacity-20 rounded-lg pl-4"
          >
            <input
              className="w-11/12 h-full outline-none py-3"
              id="full-name"
              name="full-name"
              type="text"
            />
          </div>
        </div>
        <div
          className="flex items-center my-6"
        >
          <label
            className="text-subtitle capitalize inline-block w-1/3"
            htmlFor="education-level"
          >
            education level
          </label>
          <div
            className="w-2/3 inline-flex items-center border border-black border-opacity-20 rounded-lg px-6 py-2"
          >
            <select
              className="w-full h-full outline-none py-1"
              id="education-level"
              name="education-level"
            >
              <option
                value="1"
              >
                bachelor
              </option>
              <option
                value="2"
              >
                bachelor
              </option>
              <option
                value="3"
              >
                bachelor
              </option>
              <option
                value="4"
              >
                bachelor
              </option>
            </select>
          </div>
        </div>
        <div
          className="flex items-center my-6"
        >
          <label
            className="text-subtitle capitalize inline-block w-1/3"
            htmlFor="hobbies"
          >
            hobbies
          </label>
          <div
            className="w-2/3 inline-flex items-center border border-black border-opacity-20 rounded-lg pl-4"
          >
            <input
              className="w-11/12 h-full outline-none py-3"
              id="hobbies"
              name="hobbies"
              type="text"
            />
          </div>
        </div>
        <div
          className="flex items-center my-6"
        >
          <label
            className="text-subtitle capitalize inline-block w-1/3"
            htmlFor="family-members"
          >
            family members
          </label>
          <div
            className="w-1/12 inline-flex items-center border border-black border-opacity-20 rounded-lg px-4"
          >
            <input
              className="w-full h-full outline-none py-3"
              id="family-members"
              name="family-members"
              type="text"
            />
          </div>
          <span
            className="text-subtitle capitalize mx-4"
          >
            member(s)
          </span>
        </div>
        <div
          className="flex items-center my-6"
        >
          <label
            className="text-subtitle capitalize inline-block w-1/3"
            htmlFor="gender"
          >
            gender
          </label>
          <div
            className="w-2/3 inline-flex items-center border border-black border-opacity-20 rounded-lg px-6 py-2"
          >
            <select
              className="w-full h-full outline-none py-1"
              id="gender"
              name="gender"
            >
              <option
                value="1"
              >
                Male
              </option>
              <option
                value="2"
              >
                Female
              </option>
            </select>
          </div>
        </div>
        <div
          className="flex items-center my-6"
        >
          <label
            className="text-subtitle capitalize inline-block w-1/3"
            htmlFor="birth-date"
          >
            birth date
          </label>
          <div
            className="w-2/3 inline-flex items-center border border-black border-opacity-20 rounded-lg pl-4"
          >
            <input
              className="w-11/12 h-full outline-none py-3"
              id="birth-date"
              name="birth-date"
              type="text"
            />
            <button
              type="button"
            >
              <img
                alt="profileIcon"
                src="BirthDateIcon.svg"
              />
            </button>
          </div>
        </div>
        <div
          className="flex items-center my-6"
        >
          <label
            className="text-subtitle capitalize inline-block w-1/3"
            htmlFor="email"
          >
            email
          </label>
          <div
            className="w-2/3 inline-flex items-center border border-black border-opacity-20 rounded-lg pl-4"
          >
            <input
              className="w-11/12 h-full outline-none py-3"
              id="email"
              name="email"
              type="email"
            />
          </div>
        </div>
        <div
          className="flex items-center my-6"
        >
          <label
            className="text-subtitle capitalize inline-block w-1/3"
            htmlFor="phone-number"
          >
            phone number
          </label>
          <div
            className="w-2/3 inline-flex items-center border border-black border-opacity-20 rounded-lg pl-4"
          >
            <input
              className="w-11/12 h-full outline-none py-3"
              id="phone-number"
              name="phone-number"
              type="text"
            />
          </div>
        </div>
        <div
          className="flex items-center my-6"
        >
          <label
            className="text-subtitle capitalize inline-block w-1/3"
            htmlFor="upload-id"
          >
            upload ID
          </label>
          <div
            className="w-2/3 inline-flex items-center border border-black border-opacity-20 rounded-lg pl-4"
          >
            <input
              className="w-11/12 h-full outline-none py-3"
              id="upload-id"
              name="upload-id"
              type="file"
            />
            <button
              type="button"
            >
              <img
                alt="profileIcon"
                src="Plus.svg"
              />
            </button>
          </div>
        </div>
      </form>
      <form
        className="my-16"
      >
        <h1
          className="text-title uppercase"
        >
          security
        </h1>
        <div
          className="flex items-center my-6"
        >
          <label
            className="text-subtitle capitalize inline-block w-1/3"
            htmlFor="password"
          >
            password
          </label>
          <div
            className="w-2/3 inline-flex items-center border border-black border-opacity-20 rounded-lg pl-4"
          >
            <input
              className="w-11/12 h-full outline-none py-3"
              id="password"
              name="password"
              type="password"
            />
            <button
              type="button"
            >
              <img
                alt="profileIcon"
                src="LockIcon.svg"
              />
            </button>
          </div>
        </div>
        <div
          className="flex items-center my-6"
        >
          <label
            className="text-subtitle capitalize inline-block w-1/3"
            htmlFor="confirm-password"
          >
            confirm password
          </label>
          <div
            className="w-2/3 inline-flex items-center border border-black border-opacity-20 rounded-lg pl-4"
          >
            <input
              className="w-11/12 h-full outline-none py-3"
              id="confirm-password"
              name="confirm-password"
              type="password"
            />
            <button
              type="button"
            >
              <img
                alt="profileIcon"
                src="LockIcon.svg"
              />
            </button>
          </div>
        </div>
      </form>
      <div
        className="flex justify-between my-8"
      >
        <button
          className="bg-green-300 bg-opacity-70 text-green-800 mr-4 rounded-lg uppercase px-12 py-3 text-subtitle transition duration-300 hover:bg-opacity-100 hover:shadow-lg"
          type="button"
        >
          save changes
        </button>
        <button
          className="bg-red-400 bg-opacity-70 text-red-800 mx-4 rounded-lg uppercase px-12 py-3 text-subtitle transition duration-300 hover:bg-opacity-100 hover:shadow-lg"
          type="button"
        >
          delete account
        </button>
        <button
          className="bg-yellow-300 bg-opacity-70 text-yellow-800 ml-4 rounded-lg uppercase px-12 py-3 text-subtitle transition duration-300 hover:bg-opacity-100 hover:shadow-lg"
          type="button"
        >
          cancel
        </button>
      </div>
      <div
        className="my-16"
      >
        <h1
          className="text-title capitalize mb-16"
        >
          Payment Methods & Tickets
        </h1>
        <div
          className="inline-block mr-4"
        >
          <h3
            className="text-subtitle capitalize mb-2"
          >
            3 crads added
          </h3>
          <button
            className="bg-blue-dark undefined text-black undefined rounded-lg uppercase px-12 py-3 text-subtitle transition duration-300 hover:bg-opacity-100 hover:shadow-lg"
            type="button"
          >
            show cards
          </button>
        </div>
        <div
          className="inline-block ml-4"
        >
          <h3
            className="text-subtitle capitalize mb-2"
          >
            10 tickets remaining
          </h3>
          <button
            className="bg-blue-dark undefined text-black undefined rounded-lg uppercase px-12 py-3 text-subtitle transition duration-300 hover:bg-opacity-100 hover:shadow-lg"
            type="button"
          >
            buy tickets
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
